### ISSUE TITLE
Hot Zinc Hippo - Donation Voting's reviewRecipient function may malfunction.

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2023-09-Gitcoin-judging#895]
Hot Zinc Hippo

medium

# Donation Voting's reviewRecipient function may malfunction.
## Summary
The `reviewRecipients` function in `DonationVotingMerkleDistributionBaseStrategy` sets the status of recipients in batches.
If the status value of a recipient changes immediately before executing this function, the recipient's changes will be incorrectly reverted.

## Vulnerability Details
The `reviewRecipients` function in `DonationVotingMerkleDistributionBaseStrategy`#L341-360 sets the status of recipients in batches.
In other words, not only do you change the status of the recipients you want to change, but you also set the status of all recipients in the same row (to save gas prices or simplify the logic).
```solidity
File: DonationVotingMerkleDistributionBaseStrategy.sol
341:     function reviewRecipients(ApplicationStatus[] memory statuses)
342:         external
343:         onlyActiveRegistration
344:         onlyPoolManager(msg.sender)
345:     {
346:         // Loop through the statuses and set the status
347:         for (uint256 i; i < statuses.length;) {
348:             uint256 rowIndex = statuses[i].index;
349:             uint256 fullRow = statuses[i].statusRow;
350: 
351:             statusesBitMap[rowIndex] = fullRow;
352: 
353:             // Emit that the recipient status has been updated with the values
354:             emit RecipientStatusUpdated(rowIndex, fullRow, msg.sender);
355: 
356:             unchecked {
357:                 i++;
358:             }
359:         }
360:     }
```
To do this, first read the contents of the public `statusesBitMap` variable, change the status values of the recipients you want to change there to Accepted or Rejected, group all rows containing the changed values, and call the `reviewRecipient` function.
By the modifier of `onlyActiveRegistration`, the `reviewRecipient` function call is made during the registration period. 
Therefore, if any recipient on the same row as the recipients changed between the read and write changes its status (e.g. from None to Pending) by calling `registerRecipient`, this change is transmitted by calling the `PoolOwner`'s `reviewRecipient` function (will be overwritten with None).

## Impact
Some applicants' applications were abnormally invalidated, causing the system to malfunction.

## Code Snippet
https://github.com/sherlock-audit/2023-09-Gitcoin/blob/main/allo-v2/contracts/strategies/donation-voting-merkle-base/DonationVotingMerkleDistributionBaseStrategy.sol#L341-L360

## Tool used

Manual Review

## Recommendation
The original value (statusesOld) and the new value (statusesNew) of the two parameter status values (statuses) should be passed to the `reviewRecipient` function, then the statusesOld and the current value should be compared and modified to apply statusesNew only when they are the same.
